File: src/armstrong_numbers.rs
// rank - easy
// source - exercism
pub fn is_armstrong_number(num: u32) -> bool {
    // unimplemented!("true if {} is an armstrong number", num)
    let mut num_string = String::from(num.to_string());
    let mut sum: u32 = 0;
    let len = num_string.len() as u32;

    loop {
        match num_string.pop() {
            Some(char) => sum = sum + char.to_digit(16).unwrap().pow(len),
            None => break
        }
    }

    num == sum
}

File: src/go/armstrong_numbers.go
// rank - easy
// source - exercism
package main

import (
	"fmt"
	"math"
	"strconv"
)

func armstrongNumbers(num int) bool {
	numString := strconv.FormatInt(int64(num), 10)
	sum := 0
	stringLength := len(numString)

	for i := 0; i < stringLength; i++ {
		convertedDigit, err := strconv.ParseInt(string(numString[i]), 10, 64)
		if err != nil {
			fmt.Println("Error converting string to int:", err)
		} else {
			pow := int64(math.Pow(float64(convertedDigit), float64(stringLength)))
			sum = int(int64(sum) + pow)
		}
	}

	return num == sum
}
